comm/src/test/scala/coop/rchain/comm/transport/TransportLayerRuntime.scala

```diff
@@ -163,21 +167,23 @@ abstract class TransportLayerRuntime[F[_]: Monad: Timer, E <: Environment] {
             cbl        <- createDispatcherCallback
             cb1        <- createDispatcherCallback
             cb2        <- createDispatcherCallback
-            server1 <- remoteTls1.handleReceive(
-                        protocolDispatcher.dispatch(remote1, cb1),
-                        streamDispatcher.dispatch(remote1, cb1)
-                      )
-            server2 <- remoteTls2.handleReceive(
-                        protocolDispatcher.dispatch(remote2, cb2),
-                        streamDispatcher.dispatch(remote2, cb2)
-                      )
-            r <- execute(localTl, local, remote1, remote2)
-            _ <- if (blockUntilDispatched) cb1.waitUntilDispatched()
-                else implicitly[Timer[F]].sleep(1.second)
-            _ <- if (blockUntilDispatched) cb2.waitUntilDispatched()
-                else implicitly[Timer[F]].sleep(1.second)
-            _ = server1.cancel()
-            _ = server2.cancel()
+            server1 = remoteTls1.resource(
+              protocolDispatcher.dispatch(remote1, cb1),
+              streamDispatcher.dispatch(remote1, cb1)
+            )
+            server2 = remoteTls2.resource(
+              protocolDispatcher.dispatch(remote2, cb2),
+              streamDispatcher.dispatch(remote2, cb2)
+            )
+            r <- (server1 *> server2).use { _ =>
```

Comment:
`Resource`s can be combined like any other Applicative or Monad. They will be opened and released together in specified order.